src/pandablocks_ioc/_hdf_ioc.py

```diff
@@ -332,10 +337,11 @@ def hdf_writer_names(self):
         """Formats the current dataset names for use in the HDFWriter"""
 
         hdf_names: Dict[str, Dict[str, str]] = {}
-        for record_name, dataset in self.datasets.items():
+        for record_name, dataset in self._datasets.items():
```

Comment:
It looks like the typing of the `_datasets` is not entirely accurate? If I understand its usage correctly, it's more accurately typed as a `Dict[EpicsName, Dataset]`?

Reply:
That's correct. I'll change.